src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -10747,7 +10747,7 @@ void Player_Init(Actor* thisx, PlayState* play2) {
     }
 
     if (func_80845C68(play, (respawnFlag == 2) ? 1 : 0) == 0) {
-        gSaveContext.respawn[RESPAWN_MODE_DOWN].playerParams = PARAMS_GET_S(thisx->params, 0, 8) | 0xD00;
+        gSaveContext.respawn[RESPAWN_MODE_DOWN].playerParams = PLAYER_PARAMS(PLAYER_START_MODE_IDLE, thisx->params);
```

Comment:
I guess it would be a bit more correct to wrap the second argument with`PLAYER_GET_START_BG_CAM_INDEX`
```suggestion
        gSaveContext.respawn[RESPAWN_MODE_DOWN].playerParams = PLAYER_PARAMS(PLAYER_START_MODE_IDLE, PLAYER_GET_START_BG_CAM_INDEX(thisx));
```

Reply:
I did this at first, but it doesnt match. It uses the wrong register and throws the function off. Maybe there is a way for it to match, I didnt really try much. I copied what MM currently has, since the same problem apparently existed there too.

Reply:
Oh, sounds like I probably had this very same thought and this very same issue years ago.
It's fine then

Reply:
Its pretty odd that it doesnt match, considering that it did match with `PARAMS_GET_S(thisx->params, 0, 8) | 0xD00;`
I guess it has to do with how `thisx` is being passed/accessed but I dont really know

Reply:
considered that `PLAYER_PARAMS()` already packs `startBgCamIndex` by itself, it might be expected that
```
... | PARAMS_PACK(thisx->params, 0, 8)
```
and
```
... | PARAMS_PACK(PLAYER_GET_START_BG_CAM_INDEX(thisx), 0, 8)
// or with expanded macro...
... | PARAMS_PACK(PARAMS_GET_S(thisx->params, 0, 8), 0, 8)
```
won't match

Reply:
That isnt the problem, no. It is expected that the bgcamindex is being accessed and then repacked into a new params value. That is the point of what the code is doing
The expanded form you wrote is correct and still achieves the same thing.

Reply:
odd, but ok. thanks for bothering to check it out

Reply:
@Thar0 thoughts

Reply:
I think `PARAMS_PACK_NOMASK` sounds good. I wouldn't want to do away with masking altogether since it's not really as safe from a user perspective as bits from one field could overflow into others for values that are too large

Reply:
I pushed my attempt at doing NOMASK + the change in the review comment above. Something about this is wrong but I cannot look into it right now

Reply:
`PARAMS_PACK(startMode, 8, 4)` is the important one that should be `PARAMS_PACK_NOMASK` (When I tested this I replaced both `PARAMS_PACK`s with `PARAMS_PACK_NOMASK`, but maybe it's possible to keep the bg cam one as `PARAMS_PACK`)

Reply:
(also I'm not sure if the `PARAMS_GET_U`/`PARAMS_GET_S` has anything to do with it either)